src/annotator/config/index.js

```diff
@@ -1,53 +1,240 @@
 import settingsFrom from './settings';
 import { toBoolean } from '../../shared/type-coercions';
 
+/**
+ * @typedef ConfigDefinition
+ * @prop {boolean} allowInBrowserExt - Allow this name to be available in the browser extension
+ * @prop {any} [defaultValue] - Sets a default if `valueFn` returns undefined
+ * @prop {([configName]: string) => any} valueFn -
+ *   Method to retrieve the value from the incoming source
+ * @prop {(value: any) => any} [coerce] - Transform a value's type value or value
```

Comment:
My plan was to add at least one more for highlights and move that logic out of settings. I think we could even move everything into settings and make getters for all config names or do it here.


We just need to agree on pipe line:

find the value -> validate? -> apply browser ext filter ->  apply app context filter -> set a default if needed->  coerce -> result

or something like that

Reply:
I think I'm not clear on why `coerce` would be a separate responsibility from the parsing/validating step. A default value should already be the correct type, right? I don't understand the `browserExtension` part in this when parsing/building the config. It seems like the logic could be more simplified, like:

```
for configuration property in all configuration:
  configuration property value = get value using provided valueFn callback ?? defaultValue
```

Does filtering out because `browserExtension` come into this step, or could it be a later filtering step along with the contexts?

Summary: "coercion" seems like it's part of value-parsing/possibly validation, not this module's job in the long run.